### PR TITLE
Add simple RNN in imperative

### PR DESCRIPTION
**Pull request by JiabinYang**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_imperative_split.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import paddle.fluid as fluid
+from paddle.fluid.imperative.base import to_variable
+import numpy as np
+
+
+class Split_test(fluid.imperative.Layer):
+    def __init__(self):
+        super(Split_test, self).__init__()
+
+    def _build_once(self, input):
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_imperative_split.py]: no need?

File: python/paddle/fluid/tests/unittests/test_imperative_split.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+from __future__ import print_function
+
+import unittest
+import paddle.fluid as fluid
+from paddle.fluid.imperative.base import to_variable
+import numpy as np
+
+
+class Split_test(fluid.imperative.Layer):
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_imperative_split.py]: no need? just call split directly?

File: python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+            size=[vocab_size, hidden_size],
+            dtype='float32',
+            is_sparse=False,
+            param_attr=fluid.ParamAttr(
+                name='embedding_para',
+                initializer=fluid.initializer.UniformInitializer(
+                    low=-init_scale, high=init_scale)))
+
+    def _build_once(self, input, label, init_hidden, init_cell):
+        self.softmax_weight = fluid.layers.create_parameter(
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py:164]: if you can do it in __init__, prefer __init__ over _build_once

File: python/paddle/fluid/imperative/nn.py [hunk 4]
```diff
@@ -419,3 +413,53 @@ def forward(self, input):
 
         # Currently, we don't support inplace in imperative mode
         return self._helper.append_activation(batch_norm_out)
+
+
+class EMBEDDING(layers.Layer):
```
Comment (panyx0718) [inline python/paddle/fluid/imperative/nn.py]: Embedding

File: python/paddle/fluid/imperative/nn.py [hunk 5]
```diff
@@ -419,3 +413,53 @@ def forward(self, input):
...
+    def __init__(self,
+                 size,
+                 is_sparse=False,
+                 is_distributed=False,
+                 padding_idx=None,
+                 param_attr=None,
+                 dtype='float32'):
```
Comment (panyx0718) [inline python/paddle/fluid/imperative/nn.py:462]: doc

File: python/paddle/fluid/imperative/nn.py [hunk 6]
```diff
@@ -419,3 +413,53 @@ def forward(self, input):
...
+        self._dtype = dtype
+        self._remote_prefetch = self._is_sparse and (not self._is_distributed)
+        if self._remote_prefetch:
+            assert self._is_sparse is True and self._is_distributed is False
+
+        from ..layer_helper import LayerHelper
+        self._helper = LayerHelper('embedding', param_attr=param_attr)
+
+    def _build_once(self, input):
+        self._w = self._helper.create_parameter(
```
Comment (panyx0718) [inline python/paddle/fluid/imperative/nn.py:480]: if you can do it in __init__, prefer __init__

**Review by panyx0718 — COMMENTED** [review]
some comments. mostly good

File: python/paddle/fluid/tests/unittests/test_imperative_split.py (hunk 2 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/test_imperative_split.py]: this test is used to check `MultiOutput` problem

File: python/paddle/fluid/tests/unittests/test_imperative_split.py (hunk 1 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/test_imperative_split.py]: removed

File: python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py (hunk 3 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py:164]: done

File: python/paddle/fluid/imperative/nn.py (hunk 4 above)
Comment (JiabinYang) [reply python/paddle/fluid/imperative/nn.py]: done

File: python/paddle/fluid/imperative/nn.py (hunk 5 above)
Comment (JiabinYang) [reply python/paddle/fluid/imperative/nn.py:462]: done

File: python/paddle/fluid/imperative/nn.py (hunk 6 above)
Comment (JiabinYang) [reply python/paddle/fluid/imperative/nn.py:480]: done

File: python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+                init_cell = to_variable(init_cell_data)
+                dy_loss, last_hidden, last_cell = ptb_model(x, y, init_hidden,
+                                                            init_cell)
+                if i == 0:
+                    for param in fluid.default_main_program().global_block(
+                    ).all_parameters():
+                        dy_param_init[param.name] = param._numpy()
+                dy_loss._backward()
+                sgd.minimize(dy_loss)
+                for param in fluid.default_main_program().global_block(
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py]: Can you get the parameters from ptb_model like the gan example?

Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py]: done

File: paddle/fluid/inference/utils/CMakeLists.txt [hunk 8]
```diff
@@ -1,4 +1,4 @@
 cc_library(benchmark SRCS benchmark.cc DEPS enforce)
 cc_test(test_benchmark SRCS benchmark_tester.cc DEPS benchmark)
-cc_binary(visualizer SRCS visualizer.cc DEPS analysis
-    paddle_pass_builder ir_pass_manager pass graph_viz_pass analysis_passes)
+#cc_binary(visualizer SRCS visualizer.cc DEPS analysis
```
Comment (panyx0718) [inline paddle/fluid/inference/utils/CMakeLists.txt:3]: why change this?

File: python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+                if i == 0:
+                    for param in ptb_model.parameters():
+                        dy_param_init[param.name] = param._numpy()
+                dy_loss._backward()
+                sgd.minimize(dy_loss)
+                for param in ptb_model.parameters():
+                    dy_param_updated[param.name] = param._numpy()
+                # print("dy_loss is {}".format(dy_loss._numpy()))
+                # print("last_hidden is {}".format(last_hidden._numpy()))
+                # print("last_cell is {}".format(last_cell._numpy()))
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py:273]: remove if not used?

**Review by panyx0718 — APPROVED** [review]

File: python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py (hunk 9 above)
Comment (JiabinYang) [reply python/paddle/fluid/tests/unittests/test_imperative_ptb_rnn.py:273]: done in [this PR](https://github.com/PaddlePaddle/Paddle/pull/15575)

File: paddle/fluid/inference/utils/CMakeLists.txt (hunk 8 above)
Comment (JiabinYang) [reply paddle/fluid/inference/utils/CMakeLists.txt:3]: sorry, this change will be fixed in [this PR](https://github.com/PaddlePaddle/Paddle/pull/15575)

**Pull request merged**